File: patch_app.py
import sys

with open("src/App.js", "r") as f:
    content = f.read()

# Add import
import_str = "import TopupPage from './views/TopupPage/index.js';\n"
if "import TopupPage" not in content:
    content = content.replace("import PageNotFound", import_str + "import PageNotFound")

# Add route
route_str = "<Route path=\"/topup\" element={<TopupPage />} />\n              <Route path=\"/anime\""
if "path=\"/topup\"" not in content:
    content = content.replace("<Route path=\"/anime\"", route_str)

with open("src/App.js", "w") as f:
    f.write(content)

File: patch_app.js
const fs = require('fs');
const file = 'src/App.js';
let content = fs.readFileSync(file, 'utf8');

// Add import
content = content.replace(/import JelajahiPage from '\.\/views\/JelajahiPage';/g, `import JelajahiPage from './views/JelajahiPage';\nimport KreativitasPage from './views/KreativitasPage';`);

// Add Route
content = content.replace(/<Route path="\/jelajahi" element=\{<JelajahiPage \/>\} \/>/g, `<Route path="/jelajahi" element={<JelajahiPage />} />\n            <Route path="/kreativitas" element={<KreativitasPage />} />`);

fs.writeFileSync(file, content);
